index.js

```diff
@@ -11,21 +12,36 @@ class KEAFacility extends BaseFacility {
     this._hasConf = true
     this.leases = []
     super.init()
-    this.fetchServerConf()
+    this.taskQueue = new TaskQueue(1)
   }
 
-  sendCommand (command, service, args = undefined) {
+  async _prepareLeases () {
+    if (this.taskQueue.queue.idle()) {
+      await this.fetchConf()
+      await this.fetchLeases()
+    }
+  }
+
+  setNetFac (netFac) {
+    this.netFac = netFac
+  }
+
+  async sendCommand (command, service, args = undefined) {
+    if (!this.netFac || !this.caller[this.netFac]) {
```

Comment:
netFac could be part of opts that facility accept, this way we don't need to deal with it
```
this.caller[this.opts.netFac].post
```

Reply:
update:
```
 ['fac', 'bfx-facs-http', 'c0', 'c0', { timeout: 30000, debug: false }, 0],
['fac', 'svc-facs-kea', 'k0', 'k0', () => ({ client: this.http_c0 }), 5]
```
```
this.opts.client.post
```